File: ARCADE/Intro/Level 6_3: variableName.cs
    using System;
     
    public class Test
    {
    	public static bool variableName(string name){
            for (int i = 0; i < name.Length; i++){
                if (!('a' <= name[i] && name[i] <= 'z' ||
                      'A' <= name[i] && name[i] <= 'Z' ||
                      '0' <= name[i] && name[i] <= '9' ||
                      name[i] == '_')){
                    return false;
                }
            }
            if ('0' <= name[0] && name[0] <= '9'){
                return false;
            }
            return true;
        }
    	public static void Main()
    	{
    		string  name = "var_1__Int";
    		Console.WriteLine(variableName(name));
    		string  name1 = "qq-q";
    		Console.WriteLine(variableName(name1));
    	}
    }

File: ARCADE/Intro/Level 6_3: variableName.pas
program variableName;
 
function variableName(name: string): boolean;
var i : integer;
    rs : boolean;
begin
    rs := true;
    for i := 1 to length(name) do
        if(not(((65 <= ord(name[i])) and (ord(name[i]) <= 90)) or 
               ((97 <= ord(name[i])) and (ord(name[i]) <= 122)) or 
               ((48 <= ord(name[i])) and (ord(name[i]) <= 57)) or
               (name[i] = '_'))) then rs := false;
 
    if((48 <= ord(name[1])) and (ord(name[1]) <= 57)) then rs := false;
 
    variableName := rs;
end;
 
var name : string;
begin
	name := 'var_1__Int';
	writeln(variableName(name));
	name := 'qq-q';
	writeln(variableName(name));
	name := '2w2';
	writeln(variableName(name));
end.
